test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc

```diff
@@ -762,6 +762,36 @@ class ExtProcIntegrationTest : public HttpIntegrationTest,
     }
   }
 
+  uint32_t serverReceiveBodyDuplexStreamed(absl::string_view body_sent, bool response = false,
+                                           bool compare_body = true) {
+    std::string body_received;
+    bool end_stream = false;
+    uint32_t total_req_body_msg = 0;
+    while (!end_stream) {
+      ProcessingRequest body_request;
+      EXPECT_TRUE(processor_stream_->waitForGrpcMessage(*dispatcher_, body_request));
+      if (response) {
+        EXPECT_TRUE(body_request.has_response_body()) << "Response does not have response_body \n";
+        if (compare_body) {
+          body_received = absl::StrCat(body_received, body_request.response_body().body());
+        }
+        end_stream = body_request.response_body().end_of_stream();
+      } else {
+        EXPECT_TRUE(body_request.has_request_body()) << "Request does not have request_body \n";
```

Comment:
If there is a test failure in these integration tests, the way I debug this is running the test with the trace logs enabled. And the error will be obvious by check the logs in the context.

Reply:
For example, If I deliberately cause an error here,  below are the logs are I saw. The logs give us enough info to know what's wrong: 

[2024-12-20 19:02:23.127][16][debug][testing] [./test/integration/fake_upstream.h:182] Received gRPC message: request_body {
  body: "ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss"
}
metadata_context {
}

test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc:778: Failure
Value of: body_request.has_request_body()
  Actual: true
Expected: false
Request does not have request_body 

Stack trace:
  0x3c12563: Envoy::ExtProcIntegrationTest::serverReceiveBodyDuplexStreamed()
  0x3ac81c7: Envoy::ExtProcIntegrationTest_ServerWaitForBodyBeforeSendsHeaderRespDuplexStreamed_Test::TestBody()
  0x9321bfb: testing::internal::HandleSehExceptionsInMethodIfSupported<>()
  0x9311dfa: testing::internal::HandleExceptionsInMethodIfSupported<>()
  0x92fdcd3: testing::Test::Run()
  0x92fe684: testing::TestInfo::Run()
... Google Test internal frames ...

[2024-12-20 19:02:23.127][16][debug][testing] [./test/integration/fake_upstream.h:190] Waiting for gRPC message...

Reply:
I was trying to suggest a pattern that would make test failures easier to diagnose even if you don't have deep knowledge.

It's not essential -- you could also use a debugger to figure out where the failure was. I just thought it might be easier for others without deep context to debug if you made it explicit.